test/weird-names/grammar.js

```diff
@@ -0,0 +1,16 @@
+module.exports = grammar({
+  name: "weird_names",
+  rules: {
+    program: $ => seq(
+      $.VARIABLE,
+      $.variable,
+      $._variable,
+      $.variable_
+    ),
+
+    VARIABLE: $ => '1',
```

Comment:
what happens when you lowercase and there's already a rule that use the lowercase name?

Reply:
How do you deal with the conflict?

Reply:
We already have name conflicts, for example when we remove the leading underscore to unhide a rule e.g. `_expression` -> `expression` or when add a suffix to avoid a conflict with an ocaml keyword. The example here gives the following results:
```
$ ./parse test/ok/ex1 
CST obtained from the tree-sitter parser:
program:
| variable:
| variable_:
| variable2:
| variable__:
---
Recovered typed CST:
| 1
| 2
| 3
| 4
total lines: 2
error lines: 0
error count: 0
success: 100.00%
```
So, `VARIABLE` became `variable`, `variable` became `variable_`, `_variable` became `variable2` and `variable_` became `variable__`.